sdk/appcomplianceautomation/Azure.ResourceManager.AppComplianceAutomation/src/Generated/AppComplianceReportEvidenceResource.cs

```diff
@@ -0,0 +1,442 @@
+        /// <exception cref="ArgumentNullException"> <paramref name="data"/> is null. </exception>
+        public virtual ArmOperation<AppComplianceReportEvidenceResource> Update(WaitUntil waitUntil, AppComplianceReportEvidenceData data, string offerGuid = null, string reportCreatorTenantId = null, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(data, nameof(data));
+
+            using var scope = _appComplianceReportEvidenceEvidenceClientDiagnostics.CreateScope("AppComplianceReportEvidenceResource.Update");
+            scope.Start();
+            try
+            {
+                var response = _appComplianceReportEvidenceEvidenceRestClient.CreateOrUpdate(Id.Parent.Name, Id.Name, data, offerGuid, reportCreatorTenantId, cancellationToken);
+                var uri = _appComplianceReportEvidenceEvidenceRestClient.CreateCreateOrUpdateRequestUri(Id.Parent.Name, Id.Name, data, offerGuid, reportCreatorTenantId);
+                var rehydrationToken = NextLinkOperationImplementation.GetRehydrationToken(RequestMethod.Put, uri.ToUri(), uri.ToString(), "None", null, OperationFinalStateVia.OriginalUri.ToString());
+                var operation = new AppComplianceAutomationArmOperation<AppComplianceReportEvidenceResource>(Response.FromValue(new AppComplianceReportEvidenceResource(Client, response), response.GetRawResponse()), rehydrationToken);
+                if (waitUntil == WaitUntil.Completed)
+                    operation.WaitForCompletion(cancellationToken);
+                return operation;
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Download evidence file.
+        /// <list type="bullet">
+        /// <item>
+        /// <term>Request Path</term>
+        /// <description>/providers/Microsoft.AppComplianceAutomation/reports/{reportName}/evidences/{evidenceName}/download</description>
+        /// </item>
+        /// <item>
+        /// <term>Operation Id</term>
+        /// <description>Evidence_Download</description>
+        /// </item>
+        /// <item>
+        /// <term>Default Api Version</term>
+        /// <description>2024-06-27</description>
+        /// </item>
+        /// <item>
+        /// <term>Resource</term>
+        /// <description><see cref="AppComplianceReportEvidenceResource"/></description>
+        /// </item>
+        /// </list>
+        /// </summary>
+        /// <param name="content"> Parameters for the query operation. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="content"/> is null. </exception>
+        public virtual async Task<Response<EvidenceFileDownloadResult>> DownloadAsync(EvidenceFileDownloadContent content, CancellationToken cancellationToken = default)
```

Comment:
Parameter name `EvidenceFileDownloadContent`, better to rename it to `EvidenceFileDownloadRequestContent`

Reply:
fixed